presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -396,7 +397,10 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
             return EMPTY_STATISTICS;
         }
         List<HivePartition> hivePartitions = partitionManager.getPartitions(metastore, tableHandle, constraint).getPartitions();
-        Map<String, ColumnHandle> tableColumns = getColumnHandles(session, tableHandle);
+        Map<String, ColumnHandle> tableColumns = getColumnHandles(session, tableHandle)
+                .entrySet().stream()
```

Comment:
@nezihyigitbasi The code in question was dictated to me by @electrum -- perhaps the two of you can have an offline discussion about this.